nixos/modules/services/system/nscd.nix

```
@@ -50,10 +50,22 @@ in
     systemd.services.nscd =
       { description = "Name Service Cache Daemon";
 
-        wantedBy = [ "nss-lookup.target" "nss-user-lookup.target" ];
-
         environment = { LD_LIBRARY_PATH = nssModulesPath; };
 
+        # We need system users to be resolveable in late-boot.  nscd is the proxy between
+        # nss-modules in NixOS and thus if you have nss-modules providing system users
+        # (e.g. when using DynamicUser) then nscd needs to be available before late-boot is ready
+        # We add a dependency of sysinit.target to nscd to ensure
+        # these units are started after nscd is fully started.
+        unitConfig.DefaultDependencies = false;
+        wantedBy = [
+          "sysinit.target" "nss-lookup.target" "nss-user-lookup.target"
```

@arianvp  do we even need "nss-lookup.target" "nss-user-lookup.target"?

```suggestion
          "sysinit.target"
```

---

No I think we can drop them now.   as `sysinit.target` definitely is earlier than both those units